### ISSUE TITLE
add IPC/sockets/pipes for Signals

### ISSUE THREAD

**Issue by rbouqueau** [MotionSpell/signals#2]
https://medium.com/@jamesperry/writing-high-performance-servers-in-modern-c-7cd00926828#.50i9v2sfg

or zyre like over nanomsg, or Apache Nifi...

and serialize our multimedia stack to a transport format such as ts or dtsc

**Comment by rbouqueau** [comment]
See also http://eli.thegreenplace.net/2016/grpc-sample-in-c-and-python/

**Comment by rbouqueau** [comment]
https://github.com/zaphoyd/websocketpp
http://www.zaphoyd.com/websocketpp

**Comment by rbouqueau** [comment]
On the same computer, share memory seems to be the fastest, see:
- http://www.boost.org/doc/libs/1_61_0/doc/html/interprocess.html
- http://www.boost.org/doc/libs/1_42_0/doc/html/interprocess/interprocess_smart_ptr.html

**Comment by RodolpheFouquet** [comment]
Is it easily usable with other programming languages? Maybe we could have a look on Google's Protobuf. 

https://github.com/google/protobuf

**Comment by rbouqueau** [comment]
Thanks. That's mostly for putting modules in separate execs without serializing (which implies a copy). For separate computers, we will indeed need some wider messaging + serialization.

**Comment by RodolpheFouquet** [comment]
Yeah, but if, on the same computer, we have a nodejs/go/ruby webserver controlling a signals application?

**Comment by rbouqueau** [comment]
Correct but this issue is about inter-modules communication :)

**Comment by RodolpheFouquet** [comment]
Alright then.
